mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -356,89 +263,9 @@ static ArrayAttr ldsVectorLayout(OpBuilder &b, Location loc,
-
-  SmallVector<Value, 4> globalStart = {
-      zero, coords.g_block, coords.m_block, coords.n_block, tid, zero};
-  SmallVector<Value, 4> vectorStartOuter(6, zero);
-  auto outerLoop = b.create<TransformingForOp>(
-      loc, ArrayRef<ValueRange>{globalStart, vectorStartOuter},
-      ArrayRef<Attribute>{matrixToTensor, b.getArrayAttr({})},
-      /*bounds=*/ArrayRef<int64_t>{1, 1, 1, 1, 1, dataPerThread},
-      /*strides=*/ArrayRef<int64_t>{1, 1, 1, 1, 1, vectorLen}, forceUnroll,
-      /*useIndexDiffs=*/true);
-  {
-    PatternRewriter::InsertionGuard outerGuard(b);
-    b.setInsertionPointToStart(outerLoop.getBody());
-
-    Value loaded = b.create<GlobalLoadOp>(
-        loc, loadType, tensor, outerLoop.getValidity(/*domain=*/0),
-        outerLoop.getLowerCoords(/*domain=*/0));
-
-    b.create<InBoundsStoreOp>(loc, loaded, loadBuffer,
-                              outerLoop.getLowerCoords(/*domain*/ 1)[5]);
-  }
-  return outerLoop;
-}
-
-// This function will create a DPerThread x KPerThread view of loaded register
-// buffer that may be laid out KPerThread x DPerThread or DPerThread x
-// KPerThread depending on the direction of the global vectorization.
-Value viewLoadBufferDK(PatternRewriter &b, Location loc, Value loadBuffer,
-                       GemmDimension vectorDim, int64_t copyDPerThread,
-                       int64_t copyKPerThread) {
-  SmallVector<StringRef, 2> loadBufferNames;
-  SmallVector<int64_t, 2> loadBufferShape;
-  if (vectorDim == GemmDimension::MorN) {
-    // If we are vectorizing along the M/N dimension, we have a
-    // KxD buffer that we want to transpose into a DxK buffer
-    loadBufferShape = {copyKPerThread, copyDPerThread};
-    loadBufferNames = {"k_physical", "d_physical"};
-  } else {
-    // If we are vectorizing along the K dimension, we have a
-    // DxK buffer that we want to transpose into a KxD buffer
-    loadBufferShape = {copyDPerThread, copyKPerThread};
-    loadBufferNames = {"d_physical", "k_physical"};
-  }
-  assert(loadBuffer.getType().cast<MemRefType>().getNumElements() ==
-         copyKPerThread * copyDPerThread);
-
-  Value ret;
-  BottomUpTMBuilder rawViewBuilder(b, {"rawLoad"},
-                                   {copyKPerThread * copyDPerThread});
-  rawViewBuilder.unmerge(loadBufferNames, {0, 1}, "rawLoad", loadBufferShape);
-  TransformMapAttr rawView = rawViewBuilder.get();
-  ret = b.create<TransformOp>(loc, loadBuffer, rawView);
-
-  BottomUpTMBuilder kdViewBuilder =
-      BottomUpTMBuilder::above(rawViewBuilder, rawView);
-  kdViewBuilder.passThrough({"d", "k"}, {0, 1}, {"d_physical", "k_physical"});
-  TransformMapAttr kdView = kdViewBuilder.get();
-  ret = b.create<TransformOp>(loc, ret, kdView);
-
-  return ret;
-}
-
 /// This function pack the load buffer into a store buffer ready to be copied
 /// into LDS:
-///  - The load buffer is (viewed as) a DPerThread x KPerThread
+///  - The load buffer is (viewed as) a KPerThread x DPerThread
```

Comment:
This is to be consistent with all types of views to end with ... x K x D

Reply:
Does it mean we are transposing while loading?

Reply:
no; it is a just a view of the load buffer.

Reply:
```c++
static void makeGemmInputViewsIter(TopDownTMBuilder &viewBuilder,
                                   StringRef dIterName, int64_t dPerThread,
                                   int64_t kPerThread,
                                   ArrayRef<unsigned> outDims,
                                   bool isKContigousDim) {
  if (isKContigousDim) {
    viewBuilder.merge({dIterName, "k_iter"}, outDims, "iter",
                      {dPerThread, kPerThread});
  } else {
    viewBuilder.merge({"k_iter", dIterName}, outDims, "iter",
                      {kPerThread, dPerThread});
  }
}
```
i.e. how iter is flattened will differ based on what is the contiguous dim